src/pages/workspaces/workspace/workflows/WorkflowView.js

```diff
@@ -423,17 +423,35 @@ const WorkflowView = _.flow(
   async getValidation() {
     const { namespace, name, workflowNamespace, workflowName, signal } = this.props
 
+    this.setState({ snapshotReferenceError: undefined })
     try {
       return await Ajax(signal).Workspaces.workspace(namespace, name).methodConfig(workflowNamespace, workflowName).validate()
     } catch (e) {
       if (e.status === 404) {
-        return false
+        const errmsg = await e.text()
+        // distinguish between snapshot-reference-not-found and workflow-not-found
+        if (errmsg?.includes('Reference name') && errmsg?.includes('does not exist in workspace')) {
+          this.setState(_.set(['snapshotReferenceError', errmsg]))
+          return true
+        } else {
+          return false
+        }
```

Comment:
this is kinda gross, but the back end API does not help us here. It returns 404 in cases where the bound snapshot reference isn't found as well as in cases where the workflow (method config) version isn't found. We have to parse the message itself to distinguish.

Reply:
Should we file a ticket to encapsulate this semantics in an error code returned by the relevant back-end service so we don't have to rely on message string parsing?

Reply:
filed https://broadworkbench.atlassian.net/browse/AJ-391!

Reply:
Thank you.